### DISCUSSION TITLE
To-Do list

### DISCUSSION THREAD

**Discussion by DartLazer** [DartLazer/WhoIsHomeUI#18] [Ideas]
Perhaps we can sum-up a to-do list here to create an overview, might motivate some people to work on it too!

- [ ] Rename the folders accordingly (mysite to django)
- [ ] Rename the project (mysite is to vague)
- [ ] Push the project to DockerHub once the new db seperation is fully tested and merged into main
- [ ]  Dark Theme

Added by @babanomania 
- [ ] test a multi arch build for docker, so that the docker image is compatible on arm6, arm7, x86, amd64 etc. 
- [x] add an app icon
- [ ] set an expiration for the collected data , so that after a certain number of days data can be deleted via  background jobs
- [x] set a screenshot on the main readme, so that others coming to the site can have an visual idea what the application does 
- [x] New page that shows all devices that ever connected 
- [ ] Ability to set notifications for non target devices 
- [ ] Add other notifications methods 

@babanomania @spandan13

**Comment by babanomania** [comment]
- [ ] test a multi arch build for docker, so that the docker image is compatible on arm6, arm7, x86, amd64 etc. 
- [x] add an app icon
- [ ] set an expiration for the collected data , so that after a certain number of days data can be deleted via  background jobs
- [x] set a screenshot on the main readme, so that others coming to the site can have an visual idea what the application does

**Comment by DartLazer** [comment]
That's nice! Feel free to work on some of these idea's if you want!
(for example the screenshots, I think it's a good idea!) Just pull request and I'll merge!

**Comment by babanomania** [comment]
@DartLazer i found an awesome readme template to use, should we restructure the project readme with screenshots in this structure 

https://github.com/othneildrew/Best-README-Template/blob/master/README.md

**Comment by spandan13** [comment]
Something simple like this for the logo maybe?
![20220818_225610_0000.png](https://user-images.githubusercontent.com/100998511/185458467-a9c3a08d-17e9-4a50-b85d-8a9c10fbf35b.png)

**Reply by babanomania** [reply]
This looks amazing, let's see what @DartLazer has to say about logo update . The logo can be used in the application manifest as well

**Reply by DartLazer** [reply]
> Something simple like this for the logo maybe?
> ![20220818_225610_0000.png](https://user-images.githubusercontent.com/100998511/185458467-a9c3a08d-17e9-4a50-b85d-8a9c10fbf35b.png)

Haha it's not bad at all!  Can we use it tho?

**Reply by babanomania** [reply]
@spandan13 is this logo free to use for a GPL-3 project l

**Reply by spandan13** [reply]
This logo was made by modifying a template from Canva. I checked their website for details and it says they provide a non-exclusive licence for using their templates. So unless we want to trademark the logo, we should be good to use it.

**Reply by spandan13** [reply]
> This looks amazing, let's see what @DartLazer has to say about logo update . The logo can be used in the application manifest as well

If we decide to use this logo then I can send you the svg and png files for manifest

**Reply by DartLazer** [reply]
Yes i like it lets do it!

**Reply by spandan13** [reply]
Should I keep the white background or make it transparent?

**Reply by DartLazer** [reply]
Up to you!  I like the white!

**Reply by spandan13** [reply]
Below I have uploaded icon files for the manifest @babanomania and also generated files for favicon.
[Google Drive](https://drive.google.com/drive/folders/1x5uhfvRCJaQxyF66eTZF6rdkFOUOQ_7S?usp=sharing)

**Comment by DartLazer** [comment]
> @DartLazer i found an awesome readme template to use, should we restructure the project readme with screenshots in this structure 
> 
> https://github.com/othneildrew/Best-README-Template/blob/master/README.md

I like it! If you want and have time available you can set it up if you'd like!

**Reply by babanomania** [reply]
@DartLazer check the updated readme in this branch
https://github.com/babanomania/WhoIsHomeUI/tree/readme_update

**Comment by spandan13** [comment]
Right now does the database file keep storing all info until it runs out of space?
Can we provide option to set a limit as to how much historical data it stores? Say 1 week, 1 month etc?

**Reply by DartLazer** [reply]
For now yes good idea

**Comment by DartLazer** [comment]
So logo and readme are done!

Another plan I'm gonna put on my list is a new page where users can see all the devices ever connected to the network (and delete them from the database!) perhaps here I'll also start implementing cleaning the logfiles :)

**Reply by babanomania** [reply]
Readme updates e done in the below branch,
See if you like the screenshot and the structure 

https://github.com/babanomania/WhoIsHomeUI/tree/readme_update

**Reply by DartLazer** [reply]
I love it! Merge =)? =D

**Comment by babanomania** [comment]
> So logo and readme are done!
> 
> Another plan I'm gonna put on my list is a new page where users can see all the devices ever connected to the network (and delete them from the database!) perhaps here I'll also start implementing cleaning the logfiles :) 

You can also add a drop-down in the main page for active devices and all devices,

**Comment by spandan13** [comment]
@DartLazer How about a Dark Mode toggle? ⚫

**Reply by babanomania** [reply]
Sounds interesting, but will that be a lot of change 🤔

**Reply by DartLazer** [reply]
No I think it'd be very doable.  I'll add it to my list

**Comment by DartLazer** [comment]
Just added password support to the app @babanomania and @spandan13 . How are you guys doing? any chance you could tick something off of our list =D

**Reply by spandan13** [reply]
That's a really useful addition! Will take a look in a bit. I've been using basic http auth via nginx proxy manager all this time.

**Reply by babanomania** [reply]
thats really good, let me check the dark theme feature

**Reply by DartLazer** [reply]
@babanomania thx! I was thinking we could approach this in a few ways.

-  Multiple templates dark/day (but this would require double maintenance)
- Apparently with SASS we can make a dark theme (css?) and attach it to the classes if dark is toggled (https://stackoverflow.com/questions/68667437/create-new-color-scheme-for-dark-light-mode-in-bootstrap-sass/68670276#68670276)

Any other idea's?

**Reply by DartLazer** [reply]
> That's a really useful addition! Will take a look in a bit. I've been using basic http auth via nginx proxy manager all this time.

Thx! What do you mean by that?

**Reply by babanomania** [reply]
> @babanomania thx! I was thinking we could approach this in a few ways.
> 
> -  Multiple templates dark/day (but this would require double maintenance)
> - Apparently with SASS we can make a dark theme (css?) and attach it to the classes if dark is toggled (https://stackoverflow.com/questions/68667437/create-new-color-scheme-for-dark-light-mode-in-bootstrap-sass/68670276#68670276)
> 
> Any other idea's?

Hmm using sass would be little work, the the final output would be good, I'm okay with this. 

Can you please share a sample sqlite db3 file with some dummy data to test. I'm having trouble changing colors in the chartjs chart. I've already started a new branch to investigate this.
